libs/wingc/src/type_check/symbol_env.rs

```diff
@@ -10,7 +10,7 @@ use super::UnsafeRef;
 pub struct SymbolEnv {
 	pub(crate) ident_map: HashMap<String, (StatementIdx, SymbolKind)>,
 	parent: Option<UnsafeRef<SymbolEnv>>,
-	pub return_type: Option<TypeRef>,
+	pub return_type: TypeRef,
```

Comment:
I feel like this might need some further modeling -- in some cases a SymbolEnv corresponds to an assignable block of code, like a function body (so we use this `return_type` field for type checking) but in other cases it doesn't correspond to a block of code (e.g. the environment inside a class definition). For now I inserted `void` where we previously used `None` in dummy `SymbolEnv` instances

Reply:
I don't think your change made the ambiguity any worse, but I would definitely agree it needs fourth modeling